tests/lightning_tensor/test_serialize_tensor.py

```diff
@@ -0,0 +1,936 @@
+                (hermitian_obs(np.ones(4, dtype=c_dtype), [0])),
+            ],
+        )
+
+        assert s[0] == s_expected
+
+    @pytest.mark.parametrize(
+        "test_hermobs0",
+        [(qml.Hermitian(np.eye(2), wires=[0]))],
+    )
+    @pytest.mark.parametrize(
+        "test_hermobs1",
+        [(qml.Hermitian(np.ones((2, 2)), wires=[0]))],
+    )
+    def test_hamiltonian_tensor_return(
+        self, test_hermobs0, test_hermobs1, tn_backend, use_csingle, wires_map
+    ):
+        """Test expected serialization for a tensor Hamiltonian return"""
+
+        with qml.tape.QuantumTape() as tape:
+            ham = qml.Hamiltonian(
+                [0.3, 0.5, 0.4],
+                [
+                    (test_hermobs0 @ qml.PauliY(2)),
+                    qml.PauliX(0) @ qml.PauliY(2),
+                    (test_hermobs1),
+                ],
+            )
+            qml.expval(ham @ qml.PauliZ(3))
+
+        hamiltonian_obs = get_module_name(tn_backend, "HamiltonianC", use_csingle)
+        tensor_prod_obs = get_module_name(tn_backend, "TensorProdObsC", use_csingle)
+        hermitian_obs = get_module_name(tn_backend, "HermitianObsC", use_csingle)
+        named_obs = get_module_name(tn_backend, "NamedObsC", use_csingle)
+
+        r_dtype = np.float32 if use_csingle else np.float64
+        c_dtype = np.complex64 if use_csingle else np.complex128
+
+        tn_method = tn_backend
+        tn_method = "tn" if tn_method == "exact" else tn_method
+
+        s, _ = QuantumScriptSerializer(
+            device_name, use_csingle, tensor_backend=tn_method
+        ).serialize_observables(tape, wires_map)
+
+        # Expression (ham @ obs) is converted internally by Pennylane
+        # where obs is appended to each term of the ham
+
+        s_expected = hamiltonian_obs(
+            np.array([0.3, 0.5, 0.4], dtype=r_dtype),
+            [
+                tensor_prod_obs(
+                    [
+                        (hermitian_obs(np.eye(2, dtype=c_dtype).ravel(), [0])),
+                        named_obs("PauliY", [2]),
+                        named_obs("PauliZ", [3]),
+                    ]
+                ),
+                tensor_prod_obs(
+                    [
+                        named_obs("PauliX", [0]),
+                        named_obs("PauliY", [2]),
+                        named_obs("PauliZ", [3]),
+                    ]
+                ),
+                tensor_prod_obs(
+                    [
+                        (hermitian_obs(np.ones(4, dtype=c_dtype), [0])),
+                        named_obs("PauliZ", [3]),
+                    ]
+                ),
+            ],
+        )
+
+        assert s[0] == s_expected
+
+    @pytest.mark.parametrize(
+        "test_hermobs0",
+        [(qml.Hermitian(np.eye(2), wires=[0]))],
+    )
+    @pytest.mark.parametrize(
+        "test_hermobs1",
+        [(qml.Hermitian(np.ones((2, 2)), wires=[0]))],
+    )
```

Comment:
Same here, regarding complexity. Is this necessary?

Reply:
simplified!